### PR TITLE
Key resources supplementary table

### PR DESCRIPTION
**Pull request by sjspielman**
This PR addresses [this OpenPBTA-manuscript issue](https://github.com/AlexsLemonade/OpenPBTA-manuscript/issues/217) and Key Resources table for software. Per [this comment](https://github.com/AlexsLemonade/OpenPBTA-manuscript/issues/217#issuecomment-1081288225), this table needs to be supplementary (not embedded in Methods) since there are > 10, so I'm filing in this repository where the supplementary tables are tracked.

An excel file was compiled with 4 sheets:
+ All R packages in the Dockerfile. This list was derived using [this approach](https://github.com/AlexsLemonade/OpenPBTA-manuscript/issues/217#issuecomment-1081231658). This includes all the `3.6.0` base R packages
+ All Python libraries in the Dockerfile. This list was manually compiled.
+ All command line software in the Dockerfile. This list was manually compiled.
+ All software used in the [workflows repository](https://github.com/d3b-center/OpenPBTA-workflows). This was compiled based on information in that repo's README, *but some tool versions are not provided* so they are marked "UNKNOWN" in the spreadsheet.

#### Questions for reviewers
+ I have not included the linux dependencies in the docker file, like this stuff.
```
    curl \
    bzip2 \
    zlib1g \
    libbz2-dev \
    liblzma-dev \
    libreadline-dev
```
+ Any comments on the spreadsheet organization?
+ Does anyone know where to track down the versions of tools used in the workflows? Similarly are there tools used in the workflows that are not described in the README that need to be added?

### CODE REVIEW

**Comment by jharenza** [comment]
>Does anyone know where to track down the versions of tools used in the workflows? Similarly are there tools used in the workflows that are not described in the README that need to be added?

@migbro can you please help with this? We are trying to wrap up the manuscript. Thank you!

**Comment by migbro** [comment]
Hmmm, sure can you give me a list of what is missing? I feel like most tools have a version, but in looking at those linux dependencies, that's really getting down in the weeds there. Those depend mostly on the defaults of the base image - like ubuntu18.04 vs 16.04, know what I mean?

**Comment by migbro** [comment]
I see a mention of " This was compiled based on information in that repo's README, but some tool versions are not provided so they are marked "UNKNOWN" in the spreadsheet," but I don't see that sheet.

**Comment by jharenza** [comment]
Oops, it is here: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/1f5721bdc0a50aec89e4edc0600c6e1cb20c9cf1/tables/Key_Resources.xlxs

but @sjspielman can you change the extension to `.xlsx`

**Comment by jharenza** [comment]
[Key_Resources.xlsx](https://github.com/AlexsLemonade/OpenPBTA-analysis/files/8429952/Key_Resources.xlsx)

@migbro

**Comment by sjspielman** [comment]
>but @sjspielman can you change the extension to .xlsx

Oh boy that's quite the typo I made in there copying this file over!!

**Comment by sjspielman** [comment]
Thanks, @migbro !

**Review by jharenza — APPROVED** [review]
This table looks good @sjspielman ! Thanks for adding it. 

I know the R package output is minimal code, but perhaps worth adding that to [this script ](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/tables/output_tables.Rmd) so we have the date run/output record.

I also think that we probably ought to name this file as TableSX to be consistent with the others and place it in the `results` folder. STAR methods comes at the end, so we can probably make this `TableS4`- thoughts? 

I will approve now and trust your judgement for any changes.

**Comment by sjspielman** [comment]
> I know the R package output is minimal code, but perhaps worth adding that to [this script ](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/tables/output_tables.Rmd)so we have the date run/output record.

Great idea! I'll write this to create sheet 1 and note remaining sheets are manual.

I'll change the file name for consistency, too.

**Comment by jaclyn-taroni** [comment]
I'll plan to look when the R code makes it into the table notebook. Please mention me when you're ready 😄

**Comment by sjspielman** [comment]
@jaclyn-taroni I updated this code so the table is entirely produced from the `output_tables.Rmd` notebook. A lot of this is manual data frames, but still much more reproducible than manually entered into an Excel.

File: tables/output_tables.Rmd [hunk 1]
```diff
@@ -214,6 +215,157 @@ write.xlsx(list_s3_table,
...
 
+### Create `Key Resources` table S4
+
+```{r table_s4}
+# Sheet 1: R packages in Docker image
+r_packages <- data.frame(installed.packages()[, c(1,3)]) %>%
+  rename(`R Package` = "Package")
+
+# Sheet 2: Python libraries in Docker image
+python_libraries <- tribble(
```
Comment (jaclyn-taroni) [inline tables/output_tables.Rmd:226]: For this one, could you read in and parse `requirements.txt`?

Comment (jaclyn-taroni) [reply tables/output_tables.Rmd:226]: Update: Don't think we're using that... so maybe we should remove it!

File: tables/output_tables.Rmd [hunk 2]
```diff
@@ -214,6 +215,157 @@ write.xlsx(list_s3_table,
...
+  "CrossMap","0.3.9",
+  "cycler","0.10.0",
+  "datrie","0.8.2",
+  "decorator","4.4.2",
+  "defusedxml","0.7.1",
+  "docutils","0.16",
+  "entrypoints","0.3",
+  "gitdb","4.0.7",
+  "GitPython","3.1.14",
+  "idna","2.1",
```
Comment (jaclyn-taroni) [inline tables/output_tables.Rmd:249]:
```suggestion
  "idna","2.10",
```
https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/518b5c79ace16284721f8f62d3fd989f30694938/Dockerfile#L267

File: tables/output_tables.Rmd [hunk 3]
```diff
@@ -214,6 +215,157 @@ write.xlsx(list_s3_table,
...
+)
+
+
+
+# Sheet 4: Tools used in workflows
+workflow_tools <- tribble(
+  ~"Software", ~"Version", ~"Associated Workflow(s) or Script",
+  "ControlFreeC","11.6","https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc_combined_somatic_wgs_cnv_wf.cwl",
+  "CNVKit","0.9.3","https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc_combined_somatic_wgs_cnv_wf.cwl",
+  "strekla2","2.9.3","https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc_strelka2_mutect2_manta_workflow.cwl; https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc-mutect2_strelka2-wf.cwl",
```
Comment (jaclyn-taroni) [inline tables/output_tables.Rmd:335]:
```suggestion
  "strelka2","2.9.3","https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc_strelka2_mutect2_manta_workflow.cwl; https://github.com/d3b-center/OpenPBTA-workflows/blob/master/cwl/kfdrc-mutect2_strelka2-wf.cwl",
```

**Review by jaclyn-taroni — COMMENTED** [review]
Did my best to comb through this and I uncovered a couple of issues!

File: tables/output_tables.Rmd (hunk 1 above)
Comment (sjspielman) [reply tables/output_tables.Rmd:226]: This poor unheeded line in the README 😬

Comment (sjspielman) [reply tables/output_tables.Rmd:226]: With the exceptions of the following packages (makes sense these aren't there too), `Dockerfile` and `requirements.txt` _do_ match, so I think parsing this file could be the right call.

```
setuptools	46.3.0
six	1.14.0
wheel 	0.34.2
```

This README line was (contrary to a previously deleted comment..) _definitely_ heeded.
> When adding a new package, make sure that all dependencies are also added; every package should appear with a specified version both in the Dockerfile and requirements.txt

**Comment by sjspielman** [comment]
@jaclyn-taroni Given the match between `requirements.txt` and `Dockerfile`, I ended up parsing the file directly. I think it makes sense to keep `requirements.txt` if only for others to use the file, if they want.

**Pull request merged**